services/src/main/java/org/fao/geonet/api/groups/GroupApi.java

```diff
@@ -0,0 +1,170 @@
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.FileTime;
+
+
+/**
+ * Created by juanluisrp on 24/05/2016.
+ */
+@EnableWebMvc
+@Service
+@RequestMapping(value = {
+    "/api/groups",
+    "/api/" + API.VERSION_0_1
+        + "/groups"
+})
+@Api(value = "groups",
+    tags = "groups",
+    description = "Group related operations")
+@Controller("groups")
+public class GroupApi {
+    /** API logo note.*/
+    private static final String API_GET_LOGO_NOTE = "If last-modified header "
+        + "is present it is used to check if the logo has been modified since "
+        + "the header date. If it hasn't been modified returns an empty 304 Not"
+        + " Modified response. If modified returns the image. If the group has "
+        + "no logo then returns a transparent 1x1 px PNG image.";
+
+    /** Six hours in seconds. */
+    private static final int SIX_HOURS = 60 * 60 * 6;
+    /** Transparent 1x1 px PNG encoded in Base64. */
+    private static final String TRANSPARENT_1_X_1_PNG_BASE64 = "iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR"
+        + "42mP8/5+hHgAHggJ/PchI7wAAAABJRU5ErkJggg==";
+    /** Transparent 1x1 px PNG. */
+    private static final byte[] TRANSPARENT_1_X_1_PNG = Base64.decodeBase64(TRANSPARENT_1_X_1_PNG_BASE64);
+
+
+    /**
+     * Writes the group logo image to the response. If no image is found it
+     * writes a 1x1 transparent PNG. If the request contain cache related
+     * headers it checks if the resource has changed and return a 304 Not
+     * Modified response if not changed.
+     * @param groupId the group identifier.
+     * @param webRequest the web request.
+     * @param response the servlet response.
+     * @throws ResourceNotFoundException if no group exists with groupId.
+     */
+    @ApiOperation(value = "Get the group logo image.",
+        nickname = "get",
+        notes = API_GET_LOGO_NOTE
+    )
+    @RequestMapping(value = "/{groupId}/logo", method = RequestMethod.GET)
+    public final void getGroupLogo(
+        @ApiParam(value = "Group identifier", required = true) @PathVariable(value = "groupId") final Integer groupId,
+        final WebRequest webRequest,
+        HttpServletResponse response) throws ResourceNotFoundException {
+
+        ServiceContext context = ServiceContext.get();
+        if (context == null) {
+            throw new RuntimeException("ServiceContext not available");
+        }
+
+        GroupRepository groupRepository = context.getBean(GroupRepository.class);
```

Comment:
For the i18n of the Exceptions, current implementation of [LanguageUtils.parseAcceptLanguage](https://github.com/geonetwork/core-geonetwork/blob/00e1f0b56a6c2d2573024fc8a9201fbbd30cf848/services/src/main/java/org/fao/geonet/api/tools/i18n/LanguageUtils.java#L56) doesn't work well because the three letters ISO code stored in the @[languages bean](https://github.com/geonetwork/core-geonetwork/blob/60dfc3f87019b3760700d89c45ba009d2c5e9bb9/web/src/main/webapp/WEB-INF/config-spring-geonetwork.xml#L62) don't match the ones returned by `Locale.getISO3Language`. For example, the `language`list contains `"fre"` for French, but `getISO3Language` returns `"fra"`.